### ISSUE TITLE
Restrict adding namespaces that aren't in a project for project owners/members only

### ISSUE THREAD

**Issue by gaktive** [rancher/dashboard#6594]
Internal reference: SURE-4667

The UI allows for project owners and members to add namespaces outside of the projects they're assigned to. Though this is OK with Kubernetes, within the usage of project-scoped namespaces, we're allowing users to play outside where they're supposed to.

Since project-scoped namespaces are purely UI related, there's no backend associated with that so the dashboard is doing the managing here. For project owners & project members, we need to prevent namespaces to be added outside of the project(s) they belong to. Admins can add wherever they want still.

Long term fix would rely on RBAC: https://github.com/rancher/dashboard/issues/5558

**Comment by catherineluse** [comment]
Is it also going to be restricted when you create the namespace from a workload form? Or just restricted on the Projects/Namespaces list page?

Maybe the ability to create a namespace in the workload creation form should be hidden for project members and project owners as well, until the workload form supports the ability to select a namespace for the new project. Either that or find a way to force project owners to pick a project when they are creating the namespace.

**Comment by richard-cox** [comment]
From my understanding rancher admins and cluster owners are the only two roles that should be able to create namespaces not in projects. To avoid having to fetch all global and cluster bindings to confirm this (there could be a lot of these) we could just check the `canUpdate`  / `update` link of the specific `v3/cluster`?

Once we have that disabling the 'create' button for the 'not in a project' group in the projectnamespace list should be straight forward (the group headers are custom templates in that component.

In terms of creating namespaces via the namespace selector, for this release we should hide the feature there as well. 



The namespace selector creates them outside of a project, so we would need to disable it there.

Implementation help
-

**Comment by jameson-mcghee** [comment]
@richard-cox, if I am understanding this ticket correctly, a user with the basic permissions of  "Standard User" should not be able to create namespaces that are "Not in a Project". If that is correct, that is not the functionality I am seeing on v2.6 head Commit ID db2205a. I was able to create a Namespace in that "non-project" using the "Create Namespace" button as well as by creating a new Project/Namespace and then deleting only the Project (which left the abandoned child Namespace to fall into the "Not in a Project" category).

I am setting this ticket to Reopened until those to scenarios are resolved. In addition, if you could provide me with any other specific Roles or scenarios you think should be tested for this (including specific "Built-in" roles additions such as the "User-Base" role with "Create new Clusters" or similar) that would be very helpful and appreciated. I could build a matrix and try to run through them all, but I feel like only some of these would be valid.
cc @catherineluse 

Create Namespace Button:
![image.png](https://zube.io/files/rancher/f44407d71c415485025a2578384f14f5-image.png)

Abandoned Child Namespace:
![image.png](https://zube.io/files/rancher/1b5db219175920626405f77276c3447f-image.png)
![image.png](https://zube.io/files/rancher/b594cc06080952fd83d6a9a23f97451c-image.png)
![image.png](https://zube.io/files/rancher/4202a4346a6621e87dabc4e895f76e04-image.png)

**Comment by mantis-toboggan-md** [comment]
@jameson-mcghee I saw this as well and started on a fix, I have a PR up now: https://github.com/rancher/dashboard/pull/6611

**Comment by jameson-mcghee** [comment]
@richard-cox @mantis-toboggan-md I am running a new Rancher server on v2.6-head with Commit ID cd97f6f, and I am still seeing the issues I reported previously (see screenshot). I have created an RKE2 EC2 downstream cluster (k8s v1.24.2+rke2r1) on a standard user account with 1 of each node type.
When I navigate to the Projects/Namespaces page for the standard user created cluster, I can see (and am able to add a namespace via) the "Create Namespace" button in the "Not in a Project" section.
I am also able to add namespaces to the "Not in a Project" section by creating a new project and namespace, and then deleting the project while not deleting the Namespace.
Therefore I am Reopening this ticket.
cc @sowmyav27 @catherineluse 
![image.png](https://zube.io/files/rancher/6873dc2eddd47331732b16c53c5f49ab-image.png)
